### PR TITLE
fix stubs salmon

### PR DESCRIPTION
**Pull request by Lucpen**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/salmon/index/tests/main.nf.test [hunk 1]
```diff
@@ -32,4 +32,28 @@ nextflow_process {
...
+                input[0] = Channel.of([file(params.modules_testdata_base_path + "genomics/homo_sapiens/genome/genome.fasta", checkIfExists: true)])
+                input[1] = Channel.of([file(params.modules_testdata_base_path + "genomics/sarscov2/genome/transcriptome.fasta", checkIfExists: true)])
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert path(process.out.index.get(0)).exists() },
```
Comment (nvnieuwk) [inline modules/nf-core/salmon/index/tests/main.nf.test:52]:
```suggestion
                { assert path(process.out.results.get(0)).exists() },
```

`index` is not a valid output path, I guess you want to use the `results` channel instead?

Comment (Lucpen) [reply modules/nf-core/salmon/index/tests/main.nf.test:52]: That one is for the salmon/index module, the test passes, the one that causes the issue is the nf-test for salmon/quant

File: modules/nf-core/salmon/quant/main.nf [hunk 2]
```diff
@@ -81,23 +81,6 @@ process SALMON_QUANT {
     def prefix = task.ext.prefix ?: "${meta.id}"
     """
     mkdir ${prefix}
-    mkdir ${prefix}/aux_info
```
Comment (pinin4fjords) [inline modules/nf-core/salmon/quant/main.nf:84]: Why strip all these?

File: modules/nf-core/salmon/quant/main.nf [hunk 3]
```diff
@@ -81,23 +81,6 @@ process SALMON_QUANT {
     def prefix = task.ext.prefix ?: "${meta.id}"
```
Comment (Lucpen) [inline modules/nf-core/salmon/quant/main.nf:81]:
```suggestion
    prefix = task.ext.prefix ?: "${meta.id}"
```

**Review by mashehu — APPROVED** [review]

**Pull request merged**